### PR TITLE
remove Register return value

### PR DESCRIPTION
**Pull request by NickrenREN**
Since Register() will never return err,remove the return value

### CODE REVIEW

File: daemon/container.go [hunk 1]
```diff
@@ -98,8 +98,6 @@ func (daemon *Daemon) Register(c *container.Container) error {
 
 	daemon.containers.Add(c.ID, c)
 	daemon.idIndex.Add(c.ID)
```
Comment (thaJeztah) [inline daemon/container.go]: Looking at these, it appears that we're ignoring any error that can occur when adding to the index; https://github.com/docker/docker/blob/fe9f5610c49b1a6ca2ebbb68c1478fc7f106eaa8/pkg/truncindex/truncindex.go#L59-L81

wondering if we should actually return those errors instead

/cc @cpuguy83

Comment (cpuguy83) [inline daemon/container.go]: I would say we've never checked errors here before, and the errors that could be produced are already checked earlier on before we send to truncindex.
The only really valid error is if something couldn't be inserted, but in the end is not actionable to the user, and does not affect the operation of the container in anyway, just lookup by prefix.

It probably wouldn't hurt, but it won't help either.

**Review by cpuguy83 — APPROVED** [review]
LGTM

**Review by LK4D4 — APPROVED** [review]
LGTM

File: daemon/container.go (hunk 1 above)
Comment (thaJeztah) [inline daemon/container.go]: Ok, thanks!

**Pull request merged**